builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -553,11 +553,13 @@ func resourceArmVirtualMachineCreate(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
-	osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
-	if err != nil {
-		return err
+	if _, ok := d.GetOk("os_profile"); ok {
+		osProfile, err := expandAzureRmVirtualMachineOsProfile(d)
+		if err != nil {
+			return err
+		}
+		properties.OsProfile = osProfile
 	}
```

Comment:
So I'd suggest the simplest way to achieve this would be to use a multi-phase acceptance test - we're doing this in a few places (for [example here in the AzureRM VM tests](https://github.com/hashicorp/terraform/blob/master/builtin/providers/azurerm/resource_arm_virtual_machine_test.go#L219)).

I believe the following three phases should allow us to reproduce this test-case:

1. Spin up a VM
2. Delete the VM, but keep the Disk (using the same code, without the VM resource)
3. Attach the Disk from the original VM to a new VM, using this config (without `os_profile`)

I'd expect this test might take a little while to run - so it might be worth using a `Standard_F2` size instance, rather than an A0/A1?